main.py

```diff
@@ -183,156 +185,158 @@ def compose_message(issues):
-            reply.description = "Let's curb that enthusiasm.. just a little"
-            await channel.send(embed=reply)
-        await give_random(channel, number_of_issues)
-        return
+`!feedback <N> issue/pr`
+Retrieves N least-recently-updated issue/pr in the feedback queue
 
-    elif command == "q" or command == "review":
-        if len(args) != 1:
-            await channel.send(embed=reply)
-            return
-        try:
-            number_of_issues = int(args[0].strip())
-            if number_of_issues < 1:
-                raise ValueError("Positive integer needed")
-        except ValueError:
-            await channel.send("This queue is 100% natural. Check your orders")
-            return
-        if number_of_issues > 10:
-            number_of_issues = 10
-            reply.description = "Let's curb that enthusiasm.. just a little"
-            await channel.send(embed=reply)
-
-        author = message.author
-        if command == "q":
-            author = None
-        elif isinstance(author, discord.Member):
-            author = author.nick or author.name
-        elif isinstance(author, discord.User):
-            author = author.name
-
-        await review_q(channel, number_of_issues, author)
-        return
-    return
+`!fq <N> issue/pr`
+Retrieves N least-recently-updated issue/pr in the feedback queue'''
+	await channel.send(embed=reply)
+
+
+@bot.command(name='rand')
+async def rand_cmd(ctx, number_of_issues, noun, *):
+    await check_length_issues(number_of_issues)
+    command_config=git_config["rand"]
+    update_config(noun, command_config)
+
+    await queue_handler(channel=bot.channel, number_of_issues=number_of_issues, author=None, command_config=command_config)
+
+
+@bot.command(name='review')
+async def review_cmd(ctx, number_of_issues, noun, *):
+    await check_length_issues(number_of_issues)
+    command_config=git_config["review"]
+    update_config(noun, command_config)
+
+    author = ctx.author
+    elif isinstance(author, discord.Member):
+        author = author.nick or author.name
+    elif isinstance(author, discord.User):
+        author = author.name
+
+    await queue_handler(channel=bot.channel, number_of_issues=number_of_issues, author=author, command_config=command_config)
+
+
+@bot.command(name='rq'):
+async def rq_cmd(ctx, number_of_issues, noun, *):
+    await check_length_issues(number_of_issues)
+    command_config=git_config["review"]
+    update_config(noun, command_config)
+
+    await queue_handler(channel=bot.channel, number_of_issues=number_of_issues, author=None, command_config=command_config)
+
+
+@bot.command(name='feedback')
+@bot.command(name='fq')
+async def review_cmd(ctx, number_of_issues, noun, *):
+    await check_length_issues(number_of_issues)
+    command_config=git_config["feedback"]
+    update_config(noun, command_config)
+
+    await queue_handler(channel=bot.channel, number_of_issues=number_of_issues, author=None, command_config=command_config)
```

Comment:
Let's remove the command details from `queue_handler`, put them in the commands and just keep the message sending logic there. For one-shot, we can make another function which is called in both contexts

Reply:
There isn't much of message sending logic. It defies the initial goal of code duplication reduction, since much of the code is same for all commands.